projects/plugins/crm/src/automation/commons/actions/contacts/class-add-remove-contact-tag.php

```diff
@@ -84,10 +84,7 @@ public static function get_category(): ?string {
 	 *
 	 * @throws Data_Type_Exception When the data type is not supported.
```

Comment:
Does the execute function within each action need to throw a Data_Type_Exception specifically when the data type is not supported now, since validation for the data type is happening in the validation function and that is where those exceptions are thrown? It looks like some of the actions include this still, including the highlighted example.

```suggestion
```

Reply:
Yeah, good catch.